### ISSUE TITLE
unpaved footways on rocky areas are poorly visible

### ISSUE THREAD

**Issue by matkoniecz** [gravitystorm/openstreetmap-carto#1765]
http://www.openstreetmap.org/?mlat=47.56673&mlon=12.32377#map=19/47.56673/12.32377

![](https://cloud.githubusercontent.com/assets/13821303/9294554/2e8d82a4-4451-11e5-8a5d-1c23413a1d84.png)

>But there are minor issues due to low contrast on certain background textures (i.e. scree, bare_rock) 

@geowOSM - it is better to open new issues for new issues rather than comment om closed pull requests. Commenting in open PRs is a good idea as somebody is currently working on proposal and feedback is valuable. But comments in closed PR are likely to be lost.

**Comment by imagico** [comment]
Also due to the relatively wide casing and the weak line unpaved ways look very different on dark background and on bright background which makes them difficult to recognize as the same styling across different environments.

**Comment by geowOSM** [comment]
>But comments in closed PR are likely to be lost.

Thanks for recovering my suggestion as a new issue ;)

**Comment by daganzdaanda** [comment]
I think this picture shows that unpaved paths/footways are now too weak to be useful. Even on simpler backgrounds, I think they need to be stronger.
In #1766 a progression was suggested, and that might be a solution: 
paved=solid line or long dashes, 
unknown=dot-dash, 
unpaved=dots (but strong like paved "Eggersteig" here)

**Comment by matkoniecz** [comment]
> Even on simpler backgrounds, I think they need to be stronger.

Can you give specific examples? In my testing I missed rocky areas but in many other cases it worked well (so either I missed also something else or my opinion about visibility is different).

**Comment by daganzdaanda** [comment]
Just a bit north from the example area: http://www.openstreetmap.org/#map=16/47.5840/12.3048
![unbenannt](https://cloud.githubusercontent.com/assets/7693297/9345276/8e9dfb50-460f-11e5-9770-533f1c7a4a14.GIF)
In my opinion, the unpaved paths are not visible enough on empty ground or on meadows. I always try to check how things disappear when I close my eyes halfways...
Also on farmland: http://www.openstreetmap.org/#map=17/47.83054/12.07816
![unbenannt](https://cloud.githubusercontent.com/assets/7693297/9345393/9fde932e-4610-11e5-8525-1819befea97e.GIF)

**Comment by daganzdaanda** [comment]
If I read it correctly, the line in the background has opacity 0.4:
`background/line-opacity: 0.4;`
https://github.com/matkoniecz/openstreetmap-carto/blob/9d93dda4b28eb058d913815234e53ef223bc8689/roads.mss#L1290

Would turning this to 0.8 be a solution? The casing could stay like it is. 
This should be for both paved and unpaved, unpaved might still need a bit more width than it now has.
I guess it would make the way much more visible, but maybe the contrast between white and red would be too strong?

**Comment by daganzdaanda** [comment]
Or how about showing surface with the distance of dots only, keeping the same size? The background line might need something like 0.6 opacity to show up well enough.
`line/line-dasharray: 1,3;` is used for paved at the moment. I tried to build a scale in inkscape:
![bitmap3](https://cloud.githubusercontent.com/assets/7693297/9362366/4c6772a2-46a0-11e5-9b4c-2ba796f0a8ee.png)
It seems Inkscape adds a lot of optical length when rounding a linecap. 
My impression is that all increments of 2 are resonably different. Solid > 1,2 > 1,4 > 1,6 should be okay even on real-life winding paths IMHO.

**Comment by pnorman** [comment]
> I tried to build a scale in inkscape:

It's better to do it in Mapnik. Create a file with a bunch of lines, giving each a distinct name or something, and then style them based on that name.

**Comment by pnorman** [comment]
Other styling thought - would something closer to the old highway=path line-width/dasharray work with a red colour? It might be more distinct and offer more room for variation

**Comment by SomeoneElseOSM** [comment]
@pnorman dashed vs dotted red might work - I use it on my own maps to signify wide vs narrow "England and Wales public footpaths":
 ![sample](http://i.imgur.com/hvCHgFW.png).
From memory I had to fiddle a bit with the values to make dashed and dotted red and blue have the same "visual strength" (apologies if there's a proper cartographic word for that).  The values are in the expected place in https://github.com/SomeoneElseOSM/openstreetmap-carto-AJT for info.

**Comment by kocio-pl** [comment]
@SomeoneElseOSM Looks good for me - it's much more clear than just stronger/paler red dots.

**Comment by matkoniecz** [comment]
@SomeoneElseOSM Also looks good to me, but I would prefer slightly narrower dashes.

BTW, are you maybe using also a third style - something between these two to handle missing surface values?

**Comment by SomeoneElseOSM** [comment]
@matkoniecz There are 6 "path types" used in that example - grey, blue and red (dotted and dashed each).  The colour actually represents "designation" (grey none, red "public footpath" and blue "public bridleway") and the dots means "narrow" and dashes "wide" (where "narrow" and "wide" are obtained via various other tags in lua).  The wider grey dots that you can see are abandoned railways - ignore those.

For completeness the map style (but not those two tiles) also caters for other designation values - "restricted byway", "byway open to all traffic" and various "unclassified county road" values.

**Comment by imagico** [comment]
By the way - currently footway and cycleway fill colors have strongly different lightness and chroma so they vary a lot in how heavy they appear on the map at the same line width.  In my test in https://github.com/gravitystorm/openstreetmap-carto/pull/1713#issuecomment-131612727 i adjusted the footway fill color to be stronger and therefore closer to the cycleway color (and ultimately also closer to the old path color of course in terms of lightness).  The current color (salmon, that is #FA8072) is simply too weak to be well visible at the small line widths that are necessary with the limited space available.

**Comment by drkludge** [comment]
I was asked to comment on this issue.  I think @daganzdaanda describes some
of the same problems.  I add two more examples.  I have read somewhere that
there are concerns about paved and unpaved trails and roads.  In my area of
the world the unpaved more desirable in many cases.  Strava shows how
popular my second unpaved example is
http://labs.strava.com/heatmap/#15/-112.08055/33.78501/gray/both.

I am glad to see that surface tags are being considered.  That provides a
reward for mappers to add the information.  The problem with the white
casing for unpaved or ground is that it makes the footway unusable in a
nature conservation/preserve area.  The footpaths are the most important
feature in this area.  You can see a segment of the National Trail that has
the surface=ground and another segment with nothing set for surface yet.
The two segments meet at Goat Hill.  Moreover, the National Trail is part
of the 200+ mile Maricopa Trail.  No route settings have been created in
this area yet because it is unclear how many routes are required.  I bring
the route issue up because it was mentioned in this discussion too.
http://www.openstreetmap.org/way/330787693#map=15/33.3295/-112.0901
https://www.maricopa.gov/parks/MaricopaTrail/pdf/2014maps/82k-regional-trail-south-mountain-bw-8x11.pdf

Here's another case to consider, the Sonoran Desert Preserve trails used to
be visible and useful on the OSM map.  Now the system is almost invisible
against the non-water default color for land..  The dirt trails receive
more use than the concrete trail along the road yet the bike trail is
prominently shown compared to the footway.
http://www.openstreetmap.org/#map=15/33.7830/-112.0640
As yet, I have not found a freely licensed Sonoran Desert Preserve
boundary. Otherwise, I'd have the same issue with the South Mountain
Preserve as noted above.

HTH,
Greg

**Comment by matkoniecz** [comment]
I opened "path/footway/cycleway - render missing surface separately, more prominent rendering for unpaved ways" PR #1788 - that is supposed to fix this problem.

**Comment by matkoniecz** [comment]
@SomeoneElseOSM Thanks for this rendering example!

**Comment by pnorman** [comment]
Re-opening, as it's still pretty bad on z15

z15, three unknown surface footpath
![image](https://cloud.githubusercontent.com/assets/1190866/9694088/696112e8-530a-11e5-9bc7-98335ffcdaaf.png)

I'm wondering if the bare_rock rendering needs adjustment

**Comment by imagico** [comment]
This was meant to be covered by #1793.

While the problem is most visible on bare rock the current footway styles generally give poor contrast compared to the old path rendering and the cycleway rendering.  Changing the rock pattern will not solve this.

**Comment by matkoniecz** [comment]
> I'm wondering if the bare_rock rendering needs adjustment

I thought about tweaking it, especially as z12, z13 difference is big (see http://www.openstreetmap.org/#map=12/-3.0627/37.3672 ).

**Comment by imagico** [comment]
> z12, z13 difference is big

Yes, that is because the background color is the same with and without pattern and the pattern is fairly dense.  But changing the low zoom color will likely collide with other landcover colors and changing the pattern color will make it poorly distinguishable from scree/shingle.

The majority of larger bare rock areas are fairly poor mapping anyway - it is better to primarily consider areas with more fine grained mixed mapping:

http://www.openstreetmap.org/#map=13/42.8159/-0.1251
http://www.openstreetmap.org/#map=14/46.9230/10.8245
http://www.openstreetmap.org/#map=14/57.2231/-6.2297

**Comment by hungerburg** [comment]
Unpaved footways on rocky terrain are often poorly visible on the ground too, so OSM-Carto here just mimics reality. I'd say, that is fine.
